mars/services/web/ui/src/task_info/TaskTileableGraph.js

```diff
@@ -51,39 +94,170 @@ export default class TaskTileableGraph extends React.Component {
             });
     }
 
+    fetchTileableDetail() {
+        const { sessionId, taskId } = this.props;
+
+        fetch(`api/session/${sessionId}/task/${taskId
+        }/tileable_detail`)
+            .then(res => res.json())
+            .then((res) => {
+                this.setState({
+                    tileableDetails: res,
+                })
+            });
+    }
+
     componentDidMount() {
         this.g = new dagGraphLib.Graph().setGraph({});
+
+        if (this.interval !== undefined) {
+            clearInterval(this.interval);
+        }
+        this.interval = setInterval(() => this.fetchTileableDetail(), 5000);
+        this.fetchTileableDetail();
         this.fetchGraphDetail();
     }
 
+    componentWillUnmount() {
+        clearInterval(this.interval);
+    }
+
+    /**
+     * Creates one status entry for the legend of DAG
+     *
+     * @param {*} svgContainer - The SVG container that the legend will be placed in
+     * @param {*} dotX - X coordinate of the colored dot for the legend entry
+     * @param {*} dotY - Y coordinate of the colored dot for the legend entry
+     * @param {*} textX - X coordinate of the label for the legend entry
+     * @param {*} textY - Y coordinate of the label for the legend entry
+     * @param {*} color - Status color for the legend entry
+     * @param {*} text - Label for the legend entry
+     */
+    generateGraphLegendItem(svgContainer, dotX, dotY, textX, textY, color, text) {
+        if (color === '#FFFFFF') {
+            // add an additional stroke so
+            // the white color can be visited
+            svgContainer
+                .append('circle')
+                .attr('cx', dotX)
+                .attr('cy', dotY)
+                .attr('r', 6)
+                .attr('stroke', '#333')
+                .style('fill', color);
+        } else {
+            svgContainer
+                .append('circle')
+                .attr('cx', dotX)
+                .attr('cy', dotY)
+                .attr('r', 6)
+                .style('fill', color);
+        }
+
+        svgContainer
+            .append('text')
+            .attr('x', textX)
+            .attr('y', textY)
+            .text(text)
+            .style('font-size', '15px')
+            .attr('alignment-baseline', 'middle');
+    }
+
     /* eslint no-unused-vars: ["error", { "args": "none" }] */
     componentDidUpdate(prevProps, prevStates, snapshot) {
+        if (Object.keys(this.state.tileableDetails).length !== this.state.tileables.length) {
```

Comment:
tileables may never change, but progresses of tileables may change.

Reply:
https://user-images.githubusercontent.com/69881587/130418781-824da109-5d91-4dd2-9378-7462782396bb.mov

The progress and color of the tileable successfully changed